optimize/apis-tools/optimize-api/tutorial.md

```diff
@@ -0,0 +1,162 @@
+
+:::note
+
+In this tutorial, we will execute arguments to list existing dashboard IDs and delete a dashboard. You can examine the framework for processing these arguments in the `cli.js` file before getting started.
+
+:::
+
+## GET a list of existing dashboard IDs
+
+First, let's script an API call to list our existing dashboard IDs.
+
+To do this, take the following steps:
+
+1. Examine the function `async function listDashboards([collectionId])` at the top of the `optimize.js` file. This is where you will script out your API call.
+2. Within the function, you must first apply an access token for this request:
+
+```
+const accessToken = await getAccessToken("components", optimizeAudience);
+```
+
+3. Using your generated client credentials from [prerequisites](#prerequisites), call your Optimize audience and base URL. Ensure these credentials are added to your `.env` file, and script the references to the values. For example:
+
+```
+const optimizeAudience = process.env.OPTIMIZE_AUDIENCE;
+const optimizeApiUrl = process.env.OPTIMIZE_BASE_URL;
+```
+
+4. Script the API endpoint to list the dashboard IDs of a particular collection:
+
+```
+const url = `${optimizeApiUrl}/api/public/dashboard?collectionId=${collectionId}`;
+```
+
+5. Configure your GET request to the appropriate endpoint, including an authorization header based on the previously acquired `accessToken`:
+
+```
+  const options = {
+    method: "GET",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+```
+
+6. Call the collection's endpoint, process the results from the API call, emit the dashboard IDs to output, and emit an error message from the server if necessary:
+
+```
+  try {
+    const response = await axios(options);
+    const results = response.data;
+
+    results.forEach(x => console.log(`ID: ${x.id}`));
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+```
+
+7. In your terminal, run `node cli.js optimize list` for a list of your existing dashboard IDs.
+
+:::note
+This `list` command is connected to the `listDashboards` function at the bottom of the `optimize.js` file, and executed by the `cli.js` file. While we will view dashboard IDs and delete a dashboard in this tutorial, you may add additional arguments depending on the API calls you would like to make.
+:::
+
+If you have any existing dashboards, the `ID: ${x.id}` will now output. If you have an invalid API name or action name, or no arguments provided, or improper/insufficient credentials configured, an error message will output as outlined in the `cli.js` file.
+
+## DELETE a dashboard
+
+To delete a dashboard, take the following steps:
+
+1. Outline your function, similar to the steps above:
```

Comment:
non-blocking question:

is it worth noting that the last line of this block is different, intentionally, because we'll be hitting a different endpoint this time?

Reply:
Added in latest commit 👍